src/cascadia/TerminalControl/ControlInteractivity.cpp

```diff
@@ -0,0 +1,560 @@
+            // }
+        }
+
+        _core->UpdateHoveredCell(terminalPosition);
+    }
+
+    void ControlInteractivity::TouchMoved(const winrt::Windows::Foundation::Point newTouchPoint,
+                                          const bool focused)
+    {
+        if (focused &&
+            _touchAnchor)
+        {
+            const auto anchor = _touchAnchor.value();
+
+            // Our actualFont's size is in pixels, convert to DIPs, which the
+            // rest of the Points here are in.
+            const auto fontSizeInDips{ _core->FontSizeInDips() };
+
+            // Get the difference between the point we've dragged to and the start of the touch.
+            const float dy = newTouchPoint.Y - anchor.Y;
+
+            // Start viewport scroll after we've moved more than a half row of text
+            if (std::abs(dy) > (fontSizeInDips.height<float>() / 2.0f))
+            {
+                // Multiply by -1, because moving the touch point down will
+                // create a positive delta, but we want the viewport to move up,
+                // so we'll need a negative scroll amount (and the inverse for
+                // panning down)
+                const float numRows = -1.0f * (dy / fontSizeInDips.height<float>());
+
+                const auto currentOffset = ::base::ClampedNumeric<double>(_core->ScrollOffset());
+                const auto newValue = numRows + currentOffset;
+
+                // Update the Core's viewport position, and raise a
+                // ScrollPositionChanged event to update the scrollbar
+                _updateScrollbar(newValue);
+
+                // Use this point as our new scroll anchor.
+                _touchAnchor = newTouchPoint;
+            }
+        }
+    }
+
+    void ControlInteractivity::PointerReleased(TerminalInput::MouseButtonState buttonState,
+                                               const unsigned int pointerUpdateKind,
+                                               const ::Microsoft::Terminal::Core::ControlKeyStates modifiers,
+                                               const bool /*focused*/,
+                                               const til::point terminalPosition)
+    {
+        // Short-circuit isReadOnly check to avoid warning dialog
+        if (!_core->IsInReadOnlyMode() && _canSendVTMouseInput(modifiers))
+        {
+            _trySendMouseEvent(pointerUpdateKind, buttonState, modifiers, terminalPosition);
+            return;
+        }
+
+        // Only a left click release when copy on select is active should perform a copy.
+        // Right clicks and middle clicks should not need to do anything when released.
+        const bool isLeftMouseRelease = pointerUpdateKind == WM_LBUTTONUP;
+
+        if (_core->CopyOnSelect() &&
+            isLeftMouseRelease &&
+            _selectionNeedsToBeCopied)
+        {
+            CopySelectionToClipboard(false, nullptr);
+        }
+

```

Comment:
no need for `_TryStopAutoScroll` here?

Reply:
that's still in `TermControl::_PointerReleasedHandler`